mne/utils/check.py

```diff
@@ -562,6 +562,43 @@ def _check_depth(depth, kind='depth_mne'):
     return _handle_default(kind, depth)
 
 
+def _check_dict_keys(user_options, allowed_options,
+                     user_opt_name="Channel name(s)", validator_name="info"):
+    """Check that the keys in dictionary are valid against a set list.
+
+    Return the input dictionary if it is valid,
+    otherwise raise a ValueError with a readable error message.
+
+    Parameters
+    ----------
+    user_options : dict
+        The name of the parameter to check. This is used in the error message.
+    allowed_options : list
+        All possible valid key names.
+    user_opt_name : str
+        Meaningful name of the items stored in the input dictionary.
+    validator_name : str
+        Meaningful name of the values against which the keys are
+        validated against.
+
+    Raises
+    ------
+    ValueError
+        When the key of the dict is not one of the valid options.
+
+    Returns
+    -------
+    user_dict
+        When the keys are deemed acceptable the dictionary is returned.
```

Comment:
if you accept my proposed argument renaming, then this becomes
```suggestion
    mapping
        If all keys are valid the input dict is returned unmodified.
```